src/Microsoft.AspNetCore.Server.IISIntegration/WebHostBuilderIISExtensions.cs

```diff
@@ -52,6 +52,7 @@ public static IWebHostBuilder UseIISIntegration(this IWebHostBuilder hostBuilder
                 {
                     // Delay register the url so users don't accidently overwrite it.
                     hostBuilder.UseSetting(WebHostDefaults.ServerUrlsKey, address);
+                    hostBuilder.PreferHostingUrls(true);
```

Comment:
It's only on by default for ANCM. This allows you to transparently switch from Console to IIS.